### ISSUE TITLE
Fragment events limited

### ISSUE THREAD

**Issue by iandevlin** [hakimel/reveal.js#1147]
The current events, "fragmentshown" and "fragmenthidden", that are thrown when fragments are shown or hidden, are quite limited.

I am currently in a situation when I need to know whether a previous or next fragment (related to the current one) was displayed, and the current events do not provide that information.

The addition of the direction within the fragment event would be most beneficial.

**Comment by estan** [comment]
I agree. I can't even get 'fragmentshown' to be delivered when I go backwards with Shift+Space. I'm trying to use these event to highlight some elements in an SVG figure one by one while going trough a bullet list, but it's not possible at the moment since I can't de-highlight when going backwards.
